packages/amplify_datastore/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/AmplifyDataStorePlugin.kt

```diff
@@ -144,6 +147,50 @@ class AmplifyDataStorePlugin : FlutterPlugin, MethodCallHandler {
                 }
         )
     }
+
+    @VisibleForTesting
+    fun onDelete(flutterResult: Result, request: HashMap<String, Any>) {
+        var modelName: String
+        var modelData:  HashMap<String, Any>
+        var queryPredicates: QueryPredicate;
+
+        try {
+            modelName = request["modelName"] as String
+            modelData = request["model"] as HashMap<String, Any>
+            queryPredicates = QueryPredicateBuilder.fromSerializedMap(request["queryPredicate"] as Map<String, Any>?) ?: QueryPredicates.all()
+
+        } catch (e: ClassCastException) {
+            prepareError(flutterResult, e,
+                    FlutterDataStoreFailureMessage.ERROR_CASTING_INPUT_IN_PLATFORM_CODE.toString())
+            return
+        } catch (e: Exception) {
+            prepareError(flutterResult, e,
+                    FlutterDataStoreFailureMessage.AMPLIFY_DELETE_REQUEST_MALFORMED.toString())
+            return
+        }
+
+            val plugin = Amplify.DataStore.getPlugin("awsDataStorePlugin") as AWSDataStorePlugin
```

Comment:
Ah crumbs, looks like the rest of this method is +4 too far to the right

Reply:
Sorry about that - fixed!